wrapper/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+func addProcessToCgroup(fn string, pid int) {
+	file, err := os.OpenFile(fn, os.O_WRONLY, 0644)
+	if err != nil {
+		log.Println("unable to open", fn, "for writing :", err)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	if _, err := file.WriteString(fmt.Sprintf("%d", pid)); err != nil {
+		log.Println("failed to add pid to cgroup: ", err)
+		os.Exit(1)
+	}
+}
+
+func main() {
+	// wrapper procfile cmd args...
+	if len(os.Args) < 3 {
+		log.Fatalln("too few args to wrapper")
+		os.Exit(1)
+	}
+	pfile := os.Args[1]
+	cmd := os.Args[2]
+	args := os.Args[3:]
```

Comment:
The argument with index 0 should be the program name.

```suggestion
	args := os.Args[2:]
```

Reply:
Sorry, I'll add a comment ...
```
args[0] = program
args[1] = cgroup.procs filename
args[2] = cmd
args[3:] = args
```

Reply:
Yes, but you still need to pass the command name as the 0th argument in argv to execve.

Reply:
Hmm, but I am doing that. I'm using the `LookPath` return as the `cmd` arg. Perhaps I'm still missing something here?

Reply:
@espadolini just wanted to make sure I'm addressing you concern here?

Reply:
The arguments to `syscall.Exec` are the Go equivalent to the arguments to `execve`; the path of the executable is passed in indipendently from the name passed in as `argv[0]` (and it's either illegal or at least very poor form to pass in an empty `argv`). In this case you should pass `os.Args[2:]` as the args, and `LookPath(os.Args[2])` as the cmd.